SubFixer/Model/SFConverter.m

```diff
@@ -10,27 +10,65 @@
 
 @implementation SFConverter
 
+static NSMutableArray *filesPathArray;
 static NSString *filePath;
 static NSString *folderPath;
 static NSString *fileName;
 static NSString *tempFilePath;
 
 +(void)checkDragAndFixSubtitle:(id<NSDraggingInfo>)sender {
     
-    NSArray *filesPath = [[sender draggingPasteboard] propertyListForType:NSFilenamesPboardType];
+    NSArray *pathArray = [[sender draggingPasteboard] propertyListForType:NSFilenamesPboardType];
+    NSMutableArray *filesPath = [[NSMutableArray alloc] init];
     
-    NSString *filePath = [filesPath lastObject];
     
-    NSString *fileExtension = filePath.pathExtension;
-    
-    if (![fileExtension isEqualToString:@"srt"]) {
+    for (NSString *path in pathArray) {
+        NSString *fileExtension = path.pathExtension;
         
-        [SFConverter postMessage:@"Only SRT files are supported"];
+        if (![fileExtension isEqualToString:@"srt"]) {
+            BOOL isDir;
+            BOOL exists = [[NSFileManager defaultManager] fileExistsAtPath:path isDirectory:&isDir];
+            if (exists) {
+                /* file exists */
+                if (isDir) {
+                    /* file is a directory */
+                    NSArray * dir = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:path error:NULL];
+                    for (NSString *fileName in dir) {
+                        NSString *combinedPath = [NSString stringWithFormat:@"%@%@%@", path, @"/", fileName];
```

Comment:
There is an issue here. When you find a folder and check it's content, you don't check if it's contents are srt files or not. So maybe you end up with an array full of multiple types of files.